File: code/src/nullsafety/BlackBox.java
package nullsafety;

import java.util.Random;

public class BlackBox {
    public static String foo() {
        Random random = new Random();
        if (random.nextInt() % 2 == 0) {
            return null;
        } else {
            return "Not Null";
        }
    }
}

File: code/src/nullsafety/BlackBox.kt
package nullsafety

import java.util.Random

data class Message(val text: String)

object BlackBox {
    fun getMessage(): Message? {
        val random = Random()
        
        if (random.nextInt() % 2 == 0) {
            return null
        } else {
            return Message("This is a message")
        }
    }
}
